vms/platformvm/service.go

```diff
@@ -726,44 +715,19 @@ type GetCurrentValidatorsReply struct {
 	Validators []interface{} `json:"validators"`
 }
 
-func (s *Service) loadStakerTxAttributes(txID ids.ID) (*stakerAttributes, error) {
+func (s *Service) loadStakerTxAttributes(txID ids.ID) (*state.StakerRewardAttributes, error) {
 	// Lookup tx from the cache first.
 	attr, found := s.stakerAttributesCache.Get(txID)
 	if found {
 		return attr, nil
 	}
 
 	// Tx not available in cache; pull it from disk and populate the cache.
-	tx, _, err := s.vm.state.GetTx(txID)
+	attr, err := s.vm.state.GetStakerRewardAttributes(txID)
 	if err != nil {
 		return nil, err
 	}
 
-	switch stakerTx := tx.Unsigned.(type) {
-	case txs.ValidatorTx:
-		var pop *signer.ProofOfPossession
-		if staker, ok := stakerTx.(*txs.AddPermissionlessValidatorTx); ok {
-			if s, ok := staker.Signer.(*signer.ProofOfPossession); ok {
-				pop = s
-			}
-		}
-
-		attr = &stakerAttributes{
-			shares:                 stakerTx.Shares(),
-			validationRewardsOwner: stakerTx.ValidationRewardsOwner(),
-			delegationRewardsOwner: stakerTx.DelegationRewardsOwner(),
-			proofOfPossession:      pop,
-		}
-
-	case txs.DelegatorTx:
-		attr = &stakerAttributes{
-			rewardsOwner: stakerTx.RewardsOwner(),
-		}
-
-	default:
-		return nil, fmt.Errorf("unexpected staker tx type %T", tx.Unsigned)
-	}
```

Comment:
this logic is being moved to state_helper.go and made available to process `RewardValidatorTxs`